search_single_spectrum/tools/search_single_spectrum/foodomics_track.py

```diff
@@ -46,18 +46,35 @@ def calculate_enrichment(file_occurrent_df, metadata_df):
 
     return pd.DataFrame(output_list)
 
+def metadata_file_matches(file_occurrent_df):
+    # Filter to appropriate dataset
+    file_occurrent_df = file_occurrent_df[file_occurrent_df["dataset_id"] == "MSV000084900"]
```

Comment:
I would put this line together with the file_occurrent_df["basefilename"] = .... line where you remove the extension on this dataframe into a separate method.
and call it in the main before you run the two methods:
filtered_matches = filterMatches(..)

1. you apply the same filtering in both methods and its always easier to maintain if you split the filter logic
2. with python you get weird behavior when passing variables as arguments to methods: Its either pass by reference (changes the original variable if altered in the method) or by value (changes in the method are not reflect outside). To be sure - just pass your dataframe in and return the filtered in the end

Reply:
Done